lib/src/drive-constraints.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const _ = require('lodash');
+
+module.exports = class DriveConstraints {
```

Comment:
If I understand correctly, [the function passed to `angular.service` should be a constructor](https://docs.angularjs.org/api/auto/service/$provide#service). Or you can pass a function returning an instance of the service to [`angular.factory`](https://docs.angularjs.org/api/auto/service/$provide#factory).

Even with `angular.service`, I believe it's actually a singleton, meaning `AngularJS` creates only one instance using the constructor. I'm not 100% on it. Maybe @Shou knows?

[I created a separate commit exporting an object instead of a constructor](https://github.com/resin-io/etcher/commit/f95c10ef1b17f20545ef7021b20febfef3a92aa0). If you like it more, I'll cherry pick it.